pkgs/applications/misc/syncthingtray/default.nix

```diff
@@ -0,0 +1,50 @@
+{ mkDerivation, stdenv, lib, fetchFromGitHub
+, qtbase, extra-cmake-modules, cpp-utilities, qtutilities
+, cmake, kio, plasma-framework, qttools
+, webviewProvider ? null
+, jsProvider ? null
+, enableKioPluginSupport ? false
+, enablePlasmoidSupport  ? false
+, systemdSupport ? true }:
+
+mkDerivation rec {
+  version = "0.9.1";
+  pname = "syncthingtray";
+
+  src = fetchFromGitHub {
+    owner  = "Martchus";
+    repo   = pname;
+    rev    = "v${version}";
+    sha256 = "0ijwpwlwwbfh9fdfbwz6dgi6hpmaav2jm56mzxm6as50iwnb59fx";
+  };
+
+  buildInputs = [ qtbase cpp-utilities qtutilities webviewProvider jsProvider ] 
+    ++ lib.optional enableKioPluginSupport kio
+    ++ lib.optional enablePlasmoidSupport [ extra-cmake-modules plasma-framework ]
```

Comment:
This doesn't look right. `lib.optional` only  takes one argument and implicitly wraps it in a list, so

```
   lib.optional true attribute
```

is equivalent to

```
   lib.optionals true [attribute]
```

In my opinion you should just always use `lib.optionals` and `lib.optional` should never have been added, but :man_shrugging: 
In either case, `lib.optional` with a list as the second argument is most likely wrong.

Reply:
Makes sense. I used always `lib.optionals`.